doc/reference/widgets/ChatBox.md

```diff
@@ -0,0 +1,191 @@
+# ChatBox
```

Comment:
There are is a dark themed styling issue when using the `MaterialTemplate`. Might also be with other templates. But have not tested.

![image](https://github.com/holoviz/panel/assets/42288570/89310982-5b65-4f07-b198-0819cf779c96)


```python
import panel as pn
from panel.widgets import ChatInterface, ChatEntry

pn.extension()


def even_or_odd(contents, user, instance):
    if len(contents) % 2 == 0:
        return "Even number of characters."
    return "Odd number of characters."


chat = ChatInterface(
    callback=even_or_odd,
    value=[
        ChatEntry(value="Hi", user="Assistant", avatar="🤖"),
        ChatEntry(value="I have so many questions to ask!", user="User", avatar="😊"),
    ],
)

pn.template.MaterialTemplate(main=[chat], theme="dark").servable()
```

Reply:
Fixed see other comment below
<img width="1119" alt="image" src="https://github.com/holoviz/panel/assets/15331990/19428b1f-f62d-432b-ace0-1a33cd3ca78c">